dts/arm/nordic/nrf52840.dtsi

```diff
@@ -10,6 +10,10 @@
 
 	flash0: flash {
 		reg = <0x00000000 DT_FLASH_SIZE>;
+		prop: flash-properies {
```

Comment:
Why is this a full node, rather than just putting the alignment as a property within the flash itself?

Let's make the name itself more descriptive than just "align", since it is specifically the write alignment. I don't really like the term "align", since it only implies what boundary the writes must happen on, but not that the writes have to be done in units of that size. How about something like "write-block-size".

Unsure if this should be "write-block-size", or "zephyr,write-block-size", hopefully @agross-linaro has thoughts on this.